modules/imgcodecs/src/grfmt_tiff.cpp

```diff
@@ -437,6 +440,147 @@ static void fixOrientation(Mat &img, uint16 orientation, int dst_bpp)
     }
 }
 
+static void _unpack10To16(const uchar* src, const uchar* srcEnd, ushort* dst, ushort* dstEnd, size_t expectedDstElements)
```

Comment:
> expectedDstElements

We should have both limits here (dst and src) to ensure avoiding out of buffer access.

Reply:
This is already the case.
`expectedDstElements` is the size of the currently decode tile, wothotu making assumptions on src/dst bounds limits. Then, the `fullPacketsCount` computation ensures that no OOB will occur for the "fullPackets" loop, while the last loop with "remainingElements" performs explixit bounds checking